File: ocr_runner.mjs
import { ocr } from "llama-ocr";
import fs from "fs";
import dotenv from "dotenv";

dotenv.config();

const filePath = process.argv[2];
const outputPath = process.argv[3];

if (!filePath) {
  console.error("Error: No input file path provided");
  console.log("Usage: node ocr_runner.mjs <image_path> <output_path>");
  process.exit(1);
}

if (!outputPath) {
  console.error("Error: No output file path provided");
  console.log("Usage: node ocr_runner.mjs <image_path> <output_path>");
  process.exit(1);
}

const apiKey = process.env.TOGETHER_API_KEY;
if (!apiKey) {
  console.error("Error: TOGETHER_API_KEY not found in environment variables");
  console.error("Please add your Together API key to the .env file");
  process.exit(1);
}

if (!fs.existsSync(filePath)) {
  console.error(`Error: Input file does not exist: ${filePath}`);
  process.exit(1);
}

console.log(`Processing image: ${filePath}`);
console.log(`Output will be saved to: ${outputPath}`);

try {
  const markdown = await ocr({
    filePath,
    apiKey: process.env.TOGETHER_API_KEY,
  });

  fs.writeFileSync(outputPath, markdown);
  console.log("OCR result written to:", outputPath);
  console.log(`Extracted ${markdown.length} characters`);
} catch (err) {
  console.error("Error running llama-ocr:", err);
  process.exit(1);
}

File: ocr_runner.py
import os
import subprocess
from pathlib import Path

def process_image(image_path):
    """Process an image using llama-ocr and return the extracted text."""
    # Create a temporary output file
    temp_output = Path("temp_ocr_output.md")
    
    try:
        # Run the Node.js OCR script
        print(f"Running OCR on {image_path}")
        result = subprocess.run(
            ["node", "ocr_runner.mjs", str(image_path), str(temp_output)],
            check=True,
            capture_output=True,
            text=True
        )
        
        # Print command output for debugging
        if result.stdout:
            print(f"OCR stdout: {result.stdout}")
        if result.stderr:
            print(f"OCR stderr: {result.stderr}")
        
        # Check if the output file exists
        if not temp_output.exists():
            print(f"OCR output file not created: {temp_output}")
            return None
            
        # Read the extracted text
        with open(temp_output, "r", encoding="utf-8") as f:
            extracted_text = f.read()
            
        if extracted_text:
            print(f"Successfully extracted {len(extracted_text)} characters")
            # Print a preview of the text
            print(f"Text preview: {extracted_text[:100]}...")
        else:
            print("Warning: Empty text extracted")
            
        return extracted_text
        
    except subprocess.CalledProcessError as e:
        print("Error running llama-ocr:", e)
        print(f"Return code: {e.returncode}")
        if e.output:
            print(f"Output: {e.output}")
        if e.stderr:
            print(f"Error output: {e.stderr}")
        return None
    except Exception as e:
        print(f"Unexpected error in OCR processing: {e}")
        import traceback
        traceback.print_exc()
        return None
    finally:
        # Clean up temporary file
        if temp_output.exists():
            temp_output.unlink()